contracts/extensions/bridge-operator-governance/BridgeManager.sol

```diff
@@ -0,0 +1,511 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import { SafeCast } from "@openzeppelin/contracts/utils/math/SafeCast.sol";
+import { EnumerableSet } from "@openzeppelin/contracts/utils/structs/EnumerableSet.sol";
+import { HasContracts } from "../../extensions/collections/HasContracts.sol";
+import { IQuorum } from "../../interfaces/IQuorum.sol";
+import { IBridgeManager } from "../../interfaces/IBridgeManager.sol";
+import { AddressArrayUtils } from "../../libraries/AddressArrayUtils.sol";
+import { ContractType } from "../../utils/ContractType.sol";
+import { RoleAccess } from "../../utils/RoleAccess.sol";
+import { ErrOnlySelfCall, ErrInvalidArguments, ErrLengthMismatch, ErrInvalidThreshold, ErrInvalidVoteWeight, ErrEmptyArray, ErrZeroAddress, ErrUnauthorized } from "../../utils/CommonErrors.sol";
+
+abstract contract BridgeManager is IQuorum, IBridgeManager, HasContracts {
+  using SafeCast for uint256;
+  using AddressArrayUtils for address[];
+  using EnumerableSet for EnumerableSet.AddressSet;
+
+  /// @dev value is equal to keccak256("@ronin.dpos.gateway.BridgeAdmin._governorToBridgeOperatorInfo.slot") - 1
+  bytes32 private constant GOVERNOR_TO_BRIDGE_OPERATOR_INFO_SLOT =
+    0x88547008e60f5748911f2e59feb3093b7e4c2e87b2dd69d61f112fcc932de8e3;
+  /// @dev value is equal to keccak256("@ronin.dpos.gateway.BridgeAdmin.govenorOf.slot") - 1
+  bytes32 private constant GOVENOR_OF_SLOT = 0x8400683eb2cb350596d73644c0c89fe45f108600003457374f4ab3e87b4f3aa3;
+  /// @dev value is equal to keccak256("@ronin.dpos.gateway.BridgeAdmin.governors.slot") - 1
+  bytes32 private constant GOVERNOR_SET_SLOT = 0x546f6b46ab35b030b6816596b352aef78857377176c8b24baa2046a62cf1998c;
+  /// @dev value is equal to keccak256("@ronin.dpos.gateway.BridgeAdmin.bridgeOperators.slot") - 1
+  bytes32 private constant BRIDGE_OPERATOR_SET_SLOT =
+    0xd38c234075fde25875da8a6b7e36b58b86681d483271a99eeeee1d78e258a24d;
+
+  /**
+   * @dev The domain separator used for computing hash digests in the contract.
+   */
+  bytes32 public immutable DOMAIN_SEPARATOR;
+
+  /**
+   * @dev The numerator value used for calculations in the contract.
+   */
+  uint256 internal _num;
+
+  /**
+   * @dev The denominator value used for calculations in the contract.
+   */
+  uint256 internal _denom;
+
+  /**
+   * @dev The nonce value used for tracking nonces in the contract.
+   */
+  uint256 internal _nonce;
```

Comment:
See no purpose of this variable. I think this should be used to control "one active proposal at a time", not for tracking the number of times that the thresholds are updated.